lib/px/utils/pxlogin_credentials.lua

```diff
@@ -86,6 +85,93 @@ function M.load(px_config)
         return _M.creds_extract_from_table(ci, body_json)
     end
 
+    local function split(s, sep)
+        local fields = {}
+        local sep = sep or " "
+        local pattern = string.format("([^%s]+)", sep)
+        string.gsub(s, pattern, function(c) fields[#fields + 1] = c end)
+        return fields
+    end
+
+    -- parse lines similar to:  form-data; name1=val1; name2=val2
+    local function decode_content_disposition(value)
```

Comment:
This is very similar to decoding URL query params. Maybe make this function a little more general by passing in key-value pair separator (in this case, `;`) and the key-value separator (in this case `=`) and moving it to the utils? That way it can be used to decode strings like `key=value&key2=value2` as well.

Reply:
it's a specific function to decode "Content-Disposition" line, couldn't be used anywhere else